src/plugins/comments/test/comments.e2e.js

```diff
@@ -503,6 +538,40 @@ describe('Comments', () => {
       expect(afterSetCellMetaCallback).toHaveBeenCalledWith(1, 1, 'comment', undefined, undefined, undefined);
     });
 
+    it('should not trigger `afterSetCellMeta` callback after deleting comment by context menu if `beforeSetCellMeta` returned false', () => {
+      const afterSetCellMetaCallback = jasmine.createSpy('afterSetCellMetaCallback');
+
+      handsontable({
+        data: Handsontable.helper.createSpreadsheetData(10, 10),
+        rowHeaders: true,
+        colHeaders: true,
+        contextMenu: true,
+        comments: true,
+        columns() {
+          return {
+            comment: {
+              value: 'test'
+            }
+          };
+        },
+        beforeSetCellMeta: () => false,
+        afterSetCellMeta: afterSetCellMetaCallback
+      });
+
+      expect(afterSetCellMetaCallback).not.toHaveBeenCalled();
```

Comment:
I'd remove `expect(afterSetCellMetaCallback).not.toHaveBeenCalled();` expectation and replace with visual checking if the comment was removed or not.